dashboard/app/helpers/levels_helper.rb

```diff
@@ -754,6 +754,25 @@ def lab_user_id
     Digest::SHA1.base64digest(storage_encrypt(plaintext_id)).tr('=', '')
   end
 
+  def firebase_shared_auth_token
+    return nil unless CDO.firebase_shared_secret
+
+    base_channel = params[:channel_id] || get_channel_for(@level, @user)
+    payload = {
+      uid: user_or_session_id,
+      is_dashboard_user: !!current_user,
+      channel: "#{base_channel}#{CDO.firebase_channel_id_suffix}"
+    }
+    options = {}
+    # Provides additional debugging information to the browser when
+    # security rules are evaluated.
+    options[:debug] = true if CDO.firebase_debug && CDO.rack_env?(:development)
+
+    # TODO(dave): cache token generator across requests
```

Comment:
nit: I know there are other TODOs for Dave here, but is this tracked in JIRA? At this point it would probably be a starlabs task, not something particular for Dave anyways, so might as well remove these todos and make sure we have it in the backlog

Reply:
I checked in with Dave and it seems like we should probably do the caching one but the privileges todo is more just there as an fyi. I updated the function comment to reflect that and I'll make a ticket in jira to track the caching.